exonum/src/blockchain/mod.rs

```diff
@@ -345,27 +342,34 @@ impl BlockchainMut {
         index: usize,
         fork: &mut Fork,
         tx_cache: &mut BTreeMap<Hash, Verified<AnyTx>>,
-    ) -> Result<(), Error> {
+    ) {
         let schema = Schema::new(&*fork);
         let transaction = get_transaction(&tx_hash, &schema.transactions(), &tx_cache)
-            .ok_or_else(|| format_err!("BUG: Cannot find transaction {:?} in database", tx_hash))?;
+            .unwrap_or_else(|| panic!("BUG: Cannot find transaction {:?} in database", tx_hash));
         fork.flush();
 
-        let tx_result = catch_panic(|| self.dispatcher.execute(fork, tx_hash, &transaction));
-        match &tx_result {
-            Ok(_) => {
-                fork.flush();
-            }
-            Err(e) => {
-                if e.kind == ExecutionErrorKind::Panic {
-                    error!("{:?} transaction execution panicked: {:?}", transaction, e);
-                } else {
-                    // Unlike panic, transaction failure is a regular case. So logging the
-                    // whole transaction body is an overkill: the body can be relatively big.
-                    info!("{:?} transaction execution failed: {:?}", tx_hash, e);
-                }
-                fork.rollback();
+        let tx_result = self.dispatcher.execute(fork, tx_hash, &transaction);
+        if let Err(ref err) = tx_result {
+            if err.kind() == ExecutionErrorKind::Unexpected {
+                log::error!(
+                    "{:?} transaction at {:?} resulted in unchecked error: {:?}",
+                    transaction,
+                    height,
+                    err
+                );
+            } else {
+                // Checked transaction errors are a regular occurrence. So logging the
+                // whole transaction body is an overkill: the body can be relatively big.
+                log::info!(
```

Comment:
Why not trace level? I think that a lot of failed transactions can cause node freezes.

Reply:
`info` [is used in `master` right now](https://github.com/exonum/exonum/blob/882e9c867354331ffb2867752ad1d6cc533b602b/exonum/src/blockchain/mod.rs#L373). Would you like to decrease the priority for `before` / `after_transactions` errors as well?

Reply:
IMHO `info` level is fine. Those fellows who don't want node freezes usually set the log level to `warning`/`error`, since normal node reports aren't usually that interesting.